main.rb

```diff
@@ -0,0 +1,229 @@
+# rubocop:disable Style/For
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    if instance_of?(Hash)
+      for i in 0..length - 1 do
+        yield keys[i], values[i]
+      end
+    else
+      for i in 0..to_a.length - 1 do
+        yield to_a[i]
+      end
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    if instance_of?(Hash)
+      for i in 0..length - 1 do
+        yield keys[i], values[i], i
+      end
+    else
+      for i in 0..to_a.length - 1 do
+        yield to_a[i], i
+      end
+    end
+    self
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    new_arr = to_a
+    new_arr_two = []
+    new_arr.my_each { |i| new_arr_two << i if yield i }
+    new_arr_two
+  end
+
+  def my_all?(*arg)
+    counter = true
+    if block_given?
+      my_each do |i|
+        counter = false unless yield i
+      end
+    elsif arg[0]
+      if arg[0] == Regexp
+        my_each do |i|
+          counter = false unless arg[0].match?(i)
+        end
+      elsif arg[0].is_a?(Class)
+        my_each do |i|
+          counter = false unless i.is_a?(arg[0])
+        end
+      else
+        my_each do |i|
+          counter = false unless i == arg[0]
+        end
+      end
+    else
+      my_each do |i|
+        counter = false unless i
+      end
+    end
+    counter
+  end
```

Comment:
# my_all?
- [ ] `my_all?` when a Regex is passed as an argument returns true if all of the collection matches the Regex

     Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)

       expected true
            got false